common-lib/src/main/scala/com/gu/mediaservice/model/MediaLease.scala

```diff
@@ -0,0 +1,36 @@
+package com.gu.mediaservice.model
+
+import play.api.libs.json._
+import play.api.libs.functional.syntax._
+
+import org.joda.time.DateTime
+
+
+sealed trait MediaLeaseType { def name: String }
+object MediaLeaseType {
+  implicit val reads: Reads[MediaLeaseType] = JsPath.read[String].map(MediaLeaseType(_))
+
+  implicit val writer = new Writes[MediaLeaseType] {
+    def writes(mediaLeaseType: MediaLeaseType) = JsString(mediaLeaseType.toString)
+  }
+
+  def apply(leaseType: String): MediaLeaseType = leaseType match {
+    case "allow" => AllowUseLease
+    case _ => DenyUseLease
+  }
+}
+case object AllowUseLease extends MediaLeaseType { val name = "allow" }
+case object DenyUseLease extends MediaLeaseType { val name = "deny" }
+
+case class MediaLease(
+  id: Option[String],
+  leasedBy: String,
+  startDate: Option[DateTime] = None,
```

Comment:
Because the start and end date times are optional.

No start or end means "forever" going forward, backward or both.

On Mon, 11 Apr 2016 at 13:02, James Gorrie <notifications@github.com> wrote:

> In common-lib/src/main/scala/com/gu/mediaservice/model/MediaLease.scala
> <https://github.com/guardian/grid/pull/1845#discussion_r59192675>:
>
> > +  implicit val writer = new Writes[MediaLeaseType] {
> > +    def writes(mediaLeaseType: MediaLeaseType) = JsString(mediaLeaseType.toString)
> > +  }
> > +
> > +  def apply(leaseType: String): MediaLeaseType = leaseType match {
> > +    case "allow" => AllowUseLease
> > +    case _ => DenyUseLease
> > +  }
> > +}
> > +case object AllowUseLease extends MediaLeaseType { val name = "allow" }
> > +case object DenyUseLease extends MediaLeaseType { val name = "deny" }
> > +
> > +case class MediaLease(
> > +  id: Option[String],
> > +  leasedBy: String,
> > +  startDate: Option[DateTime] = None,
>
> Why would these be options?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/grid/pull/1845/files/9e7b6ca507b1a4040c188263bc1d14b33aa6d83e#r59192675>
>

-- 

------------------------------
This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396